src/error.rs

```diff
@@ -0,0 +1,24 @@
+use super::ColoredString;
+use std::{error::Error, fmt};
+
+pub struct ColoredStringError(pub ColoredString);
```

Comment:
may I ask what does it mean if a struct has a parameter like pub ColoredString ?

I usually see creation of struct with something like
`struct NameOfStruct {}` with curly braces, not with parenthesis. I'm fairly new to Rust as well. I haven't finished the Rust book but I'm on Ch 13 and consistently going back to older chapters.

Reply:
That means it's a [tuple struct](https://doc.rust-lang.org/1.9.0/book/structs.html#tuple-structs). I'd say it's most useful when you're wrapping a type to modify it's behavior, and you don't need it to have named fields. I could have defined it like this:
```rust
pub struct ColoredStringError {
    pub colored_string: ColoredString,
}
```
But I felt that that would be overly verbose.